### PR TITLE
[1.0?] Fix AutoMM warnings

### PR DESCRIPTION
**Pull request by FANGAreNotGnu**
- Refactor warning filters and add one more filter for object detection
- Remove messaging in DDP subprocesses

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/learners/matching.py [hunk 1]
```diff
@@ -221,6 +221,8 @@ def __init__(
         self._total_train_time = None
         self._best_score = None
 
+        self._log_filters = []
```
Comment (zhiqiangdon) [inline multimodal/src/autogluon/multimodal/learners/matching.py:224]: Matcher shares the same `run_trainer` API with base learner. Using an empty log filter is overwriting the log filters in base learner.

Comment (FANGAreNotGnu) [reply multimodal/src/autogluon/multimodal/learners/matching.py]: Fixed with correct filters. Matcher did not call `super.__init__`, so we'll need to initialize `self._log_filters`.

**Review by zhiqiangdon — APPROVED** [review]
LGTM!

**Review by zhiqiangdon — COMMENTED** [review]
The logs in the built docs have issues.

**Review by zhiqiangdon — COMMENTED** [review]
Looks like `from pytorch_lightning.utilities import rank_zero_only` makes our `apply_log_filter` can't work properly. See http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-3770/ef6637e/tutorials/multimodal/text_prediction/beginner_text.html#training 
So, I removed this import.

**Comment by FANGAreNotGnu** [comment]
> Looks like `from pytorch_lightning.utilities import rank_zero_only` makes our `apply_log_filter` can't work properly. See http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-3770/ef6637e/tutorials/multimodal/text_prediction/beginner_text.html#training So, I removed this import.

I have no idea why `rank_zero_only` breaks our `apply_log_filter`. Let me check their source code. And we might need to remove ddp subprocess warning after 1.0 release.

**Pull request merged**